tests/test_brokertypes.py

```diff
@@ -0,0 +1,473 @@
+        self.assertNotEqual(ZeekEvent('Test::event', from_py('hello'), from_py(42), from_py(True)),
+                            ZeekEvent('Test::event2', from_py('hello'), from_py(42), from_py(True)))
+        self.assertNotEqual(ZeekEvent('Test::event', from_py('hello'), from_py(42), from_py(True)),
+                            ZeekEvent('Test::event', from_py('hello'), from_py(42)))
+        self.assertNotEqual(ZeekEvent('Test::event', from_py('hello'), from_py(43)),
+                            ZeekEvent('Test::event', from_py('hello'), from_py(42)))
+
+        self.assertEqualRoundtrip(evt)
+
+        vec = Vector.unserialize(evt.serialize())
+        evt2 = ZeekEvent.from_vector(vec)
+        self.assertEqual(evt, evt2)
+
+        with self.assertRaises(TypeError):
+            ZeekEvent.from_vector(String('not a vector'))
+
+        with self.assertRaises(TypeError):
+            ZeekEvent('foo', 1)
+
+    def test_handshake_message(self):
+        self.assertEqual(HandshakeMessage(['foo', 'bar']),
+                         HandshakeMessage(['foo', String('bar')]))
+        self.assertNotEqual(HandshakeMessage(['foo', 'bar']),
+                            HandshakeMessage(['foo']))
+        self.assertEqualRoundtrip(HandshakeMessage(['foo', 'bar']))
+
+    def test_handshake_ack_message(self):
+        self.assertEqual(HandshakeAckMessage('aaaa', '1.0'),
+                         HandshakeAckMessage('aaaa', '1.0'))
+        self.assertNotEqual(HandshakeAckMessage('aaaa', '1.0'),
+                            HandshakeAckMessage('bbbb', '1.0'))
+        self.assertEqualRoundtrip(HandshakeAckMessage('aaaa', '1.0'))
+
+    def test_data_message(self):
+        self.assertEqual(DataMessage('foo', String('test')),
+                         DataMessage('foo', String('test')))
+        self.assertNotEqual(DataMessage('foo', String('test')),
+                            DataMessage('foo', String('other')))
+        self.assertEqualRoundtrip(DataMessage('foo', String('test')))
+
+    def test_error_message(self):
+        msg1 = ErrorMessage('deserialization_failed', 'this is where you failed')
+        msg2 = ErrorMessage('deserialization_failed', 'this is where you also failed')
+        self.assertEqual(msg1, msg1)
+        self.assertNotEqual(msg1, msg2)
+        self.assertEqualRoundtrip(msg1)
+
+    def test_type_lt(self):
+        # Any brokertyped data value can be compared to any other, but not to
+        # unrelated types.
+        self.assertTrue(Address('127.0.0.1') < Boolean(True))
+        self.assertTrue(Boolean(True) < Count(1))
+        self.assertTrue(Count(1) < Integer(1))
+        self.assertTrue(Integer(1) < Real(1))
+        self.assertTrue(Real(1) < Set())
+        self.assertTrue(Set() < String('foo'))
+        self.assertTrue(String('foo') < Subnet('10.0.0.0/8'))
+        self.assertTrue(Subnet('10.0.0.0/8') < Timespan('10d'))
+        self.assertTrue(Timespan('10d') < Timestamp(datetime.datetime.now()))
```

Comment:
Is this primarily for sortability so that one can store these in heterogenous containers (in contrast to having some actual semantic meaning)?

Reply:
Exactly. If you take the Broker data model pretty literally, then vectors, tables, etc can store heterogeneous types, and being able to sort here helps to make representations consistent, which helps for testing/troubleshooting/etc.